### PR TITLE
ELO-59: Prepare for Beta

### PR DESCRIPTION
**Pull request by aturingmachine**
#### Description
The `main` process now opens the serial port to the card reader, and is hooked into the sign in flow that already existed. Also added in some pillows in case the card reader is disconnected like health checks.

#### Technical Details
- Updates `main/index.js` to house the serialport logic
- Adds serialport reading to existing "sign in" flow
- Adds health checks and error handling for a failing card reader connection

#### Notes
- This needs to be thoroughly tested. It is a Sunday and I just spent like 3 hours in dependency versioning hell to get this working so I didn't test it *that* much...

### CODE REVIEW

File: kiosk/src/main/index.js [hunk 1]
```diff
@@ -1,62 +1,153 @@
...
+    }
 
-  mainWindow.loadURL(winURL)
+    console.log(ports);
 
-  mainWindow.on('closed', () => {
-    mainWindow = null
-  })
+    scannerPort = ports.find(port => {
+      return port.comName === "/dev/ttyUSB0";
```
Comment (dnjhnsn52) [inline kiosk/src/main/index.js]: Will this scanner always be on this dev address?

Comment (dnjhnsn52) [reply kiosk/src/main/index.js]: Or was the device id I was using not working with your HID?

Comment (aturingmachine) [reply kiosk/src/main/index.js]: I can look into this more as using `/dev/ttyUSB0` is far from stable. I will try something out this weekend before I start work on the actual Kiosk

Comment (dnjhnsn52) [reply kiosk/src/main/index.js]: 👍

**Pull request merged**